File: baekjoon/stage16/1149_RGB거리.py
import sys

def calculate(n):
    min_val = 0
    if n == 1:
        min_val = min(cost[0][0], cost[0][1], cost[0][2])
    min_costs = [[cost[0][0], cost[0][1], cost[0][2]]]
    for i in range(1, n):
        min_costs.append([min(min_costs[i-1][1], min_costs[i-1][2])+cost[i][0],
                          min(min_costs[i-1][0], min_costs[i-1][2])+cost[i][1],
                          min(min_costs[i-1][0], min_costs[i-1][1])+cost[i][2]])
    min_val = min(min_costs[n-1][0], min_costs[n-1][1], min_costs[n-1][2])
    return min_val

if __name__ == "__main__":
    n = int(sys.stdin.readline())

    cost = []
    for i in range(n):
        a, b, c = map(int, sys.stdin.readline().split())
        cost.append([a, b, c])

    min_val = calculate(n)
    print(min_val)

File: problem-solving/baekjoon/1149_RGB거리.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscanln(reader, &n)

	cost := [][]int{}
	for i := 0; i < n; i++ {
		var a, b, c int
		fmt.Fscanln(reader, &a, &b, &c)
		cost = append(cost, []int{a, b, c})
	}
	minVal := calculate(cost, n)
	fmt.Fprintln(writer, minVal)
}

func calculate(cost [][]int, n int) (minVal int) {
	minVal = 0
	if n == 1 {
		minVal = int(math.Min(math.Min(float64(cost[0][0]), float64(cost[0][1])), float64(cost[0][2])))
	}

	minCosts := [][]int{}
	minCosts = append(minCosts, []int{cost[0][0], cost[0][1], cost[0][2]})

	for i := 1; i < n; i++ {
		values := []int{
			int(math.Min(float64(minCosts[i-1][1]), float64(minCosts[i-1][2]))) + cost[i][0],
			int(math.Min(float64(minCosts[i-1][0]), float64(minCosts[i-1][2]))) + cost[i][1],
			int(math.Min(float64(minCosts[i-1][0]), float64(minCosts[i-1][1]))) + cost[i][2],
		}
		minCosts = append(minCosts, values)
	}

	minVal = int(math.Min(math.Min(float64(minCosts[n-1][0]), float64(minCosts[n-1][1])), float64(minCosts[n-1][2])))
	return
}
